include/deal.II/base/derivative_form.h

```diff
@@ -366,50 +370,76 @@ DerivativeForm<order, dim, spacedim, Number>::memory_consumption()
 
 
 /**
- * One of the uses of DerivativeForm is to apply it as a transformation. This
- * is what this function does.  If @p DF is DerivativeForm<1,dim,1> it computes
- * $DF * T$, if @p DF is DerivativeForm<1,dim,rank> it computes $T*DF^{T}$.
+ * One of the uses of DerivativeForm is to apply it as a linear transformation.
+ * This function returns $\nabla \mathbf F(\mathbf x) \Delta \mathbf x$, which
+ * approximates the change in $\mathbf F(\mathbf x)$ when $\mathbf x$ is changed
+ * by the amount $\Delta \mathbf x$
+ * @f[
+ *   \nabla \mathbf F(\mathbf x) \; \Delta \mathbf x
+ *   \approx
+ *   \mathbf F(\mathbf x + \Delta \mathbf x) - \mathbf F(\mathbf x).
+ * @f]
+ * The transformation corresponds to
+ * @f[
+ *   [\text{result}]_{i_1,\dots,i_k} = i\sum_{j}
+ *   \left[\nabla \mathbf F(\mathbf x)\right]_{i_1,\dots,i_k, j}
+ *   \Delta x_j
+ * @f]
+ * in index notation and corresponds to
+ * $[\Delta \mathbf x] [\nabla \mathbf F(\mathbf x)]^T$ in matrix notation.
  *
  * @relatesalso DerivativeForm
- * @author Sebastian Pauletti, 2011
+ * @author Sebastian Pauletti, 2011, Reza Rastak, 2019
  */
 template <int spacedim, int dim, typename Number>
 inline Tensor<1, spacedim, Number>
-apply_transformation(const DerivativeForm<1, dim, spacedim, Number> &DF,
-                     const Tensor<1, dim, Number> &                  T)
+apply_transformation(const DerivativeForm<1, dim, spacedim, Number> &grad_F,
+                     const Tensor<1, dim, Number> &                  d_x)
 {
   Tensor<1, spacedim, Number> dest;
   for (unsigned int i = 0; i < spacedim; ++i)
-    dest[i] = DF[i] * T;
+    dest[i] = grad_F[i] * d_x;
   return dest;
 }
 
 
 
 /**
- * Similar to previous apply_transformation. It computes $T*DF^{T}$.
+ * Similar to the previous apply_transformation().
+ * Each row of the result corresponds to one of the rows of @p D_X transformed
+ * by @p grad_F, equivalent to $\text{D\_X} * \text{grad\_F}^T$ in matrix notation.
```

Comment:
```suggestion
 * by @p grad_F, equivalent to $\text{D\_X}  \text{grad\_F}^T$ in matrix notation.
```

Reply:
Not sure what the most consistent notation is. I avoid using asterisks in tensor equations but use asterisk in matrix formulas. In the documentation of FullMatrix<Number>, for example in the mmult member function, asterisk is used. What is your opinion?

Reply:
Some of these cases date back to a time when we could not render formulas properly in the documentation. These days, we try to use mathematical notation, which does not use the asterisk. Every once in a while we come by cases where we still use them and then someone fixes it. (So, if you feel like it, write a separate pull request to remove the asterisks in other places ;-) )

Reply:
Got it! thanks :)